tools/apiview/parsers/CONTRIBUTING.md

```diff
@@ -0,0 +1,103 @@
+
+- `Name (n)`  : The name of the tree node which will be used as label for the API Navigation. Generally use the name of the module (class, method)
+- `Id (i)` : Id of the node, which should be unique at the node level. i.e. unique among its siblings. Use whatever existing parser is assigning to DefinitionId for the main Token of the node. Each node must have an Id.
+- `Kind (k)` : What kind of node is it. Please ensure you set a Kind for each node. Using any of the following `assembly`, `class`, `delegate`, `enum`, `interface`, `method` , `namespace`, `package`, `struct`, `type` will get you the corresponding default icons for the page navigation but feel free to use something language specific.
+- `Tags (t)` : Use this for opt in or opt out boolean properties. The currently supported tags are
+  - `Deprecated` Mark a node as deprecated
+  - `Hidden`  Mark a node as Hidden
+  - `HideFromNavigation` Indicate that anode should be hidden from the page navigation.
+  - `SkipDiff` Indicate that a node should not be used in computatation of diff.
+  - `CrossLandDefId` The cross language definitionId for the node.
+- `Properties (p)` : Use this for other properties of the node. The currently supported tags
+  - `SubKind` Similar to kind, use this to make the node more specific. e.g.   `Kind = 'Type'`, and `SubKind = 'class'` or somethign specific to you language. We also use this to make the navigation icon it will overide kind.
+  - `IconName` Use this only if you are looking to add a custom icon different from language wide defaults.
+- `TopTokens (tt)` : The main data of the node. This is all the tokens that actually define the node. e.g. For a class this would include the access modifier, the class name, any attributes or decorators e.t.c. For a method this would include the return type, method name, parameters e.t.c. See StructureTokesn description below for more info.
+- `BottomToken (bt)` : Data that closes out the node. Depending on the language this would include the closing curly brace and/or empty lines. You can simulate an empty line by adding an empty token (content token with no value) and a lineBreak token.
+- `Children (c)` : The nodes immediate children. For a namespace this would be classes, for a class this would be the class constructors and methods.
+
+### StructuredToken
+- `Value (v)` : The token value which will be dispalyed. Can be null.  Spacing tokens don't need to have value.
+- `Id (i)` : This is essentially whatever existing parser was asigning to the token DefinitionId. You dont have to assign a tokenId.
+- `Kind (k)` : An enum
+  - `Content` Specifies that the token is content
+  - `LineBreak` Space token indicating switch to new line.
+  - `NoneBreakingSpace` Regular single space
+  - `TabSpace` 4 NoneBreakingSpaces
+  - `ParameterSeparator` Use this between method parameters. Depending on user setting this would result in a singlespace or new line
```

Comment:
Please explain. In v1, this wasn't a thing and I would expect that stuff to be `topToken`s of whatever parameter I'm outputting. What "user setting" are you talking about?

Reply:
You are right this is a new concept needed to support a new request. In the default case this will be the same a s none breaking `NonBreakingSpace`. But it could also be a `LineBreak`, or `LineBreak`  + `TabSpace`  as is the case with your requirement. This is just to let us support the different scenarios and be able to switch from one to the other as required.